### DISCUSSION TITLE
Do R-universe packages get checked with other R-universe packages?

### DISCUSSION THREAD

**Discussion by wlandau** [r-universe-org/help#371] [General]
This may be a naive question, but when R-universe checks a package, does it get the dependencies from the same universe? In other words, do the dependencies in the universe supersede dependencies on CRAN in R CMD check?

**Comment by jeroen** [comment]
If you look at the raw build log for a package check, you can see where each dependency is being downloaded from.

The current universe is indeed automatically added to the front of `options("repos")`, so in most cases packages will be downloaded from the current universe if available there.

However I think R always tries to download the highest version of the package. So if the version on CRAN is greater than the one in the current universe, `install.packages()` may still download the CRAN version instead. But I am not 100% sure if this is still the case (the way `install.packages()` resolves dependencies is very messy, `pak` may do a better job).

**Reply by wlandau** [reply]
Ah, thanks for the tip. Indeed, when I look at https://github.com/r-universe/r-releases/actions/runs/8124901191/job/22206916128#step:5:139 I see that the check is trying to look in the same universe for dependencies. This is ideal for `r-releases`.

> However I think R always tries to download the highest version of the package. So if the version on CRAN is greater than the one in the current universe, install.packages() may still download the CRAN version instead. But I am not 100% sure if this is still the case (the way install.packages() resolves dependencies is very messy, pak may do a better job).

Sounds messy, and I hope `pak` can help pull from different repos in a more consistent way, but this is not a show-stopper for `r-releases` by any means.
